### PR TITLE
New feature added

### PR DESCRIPTION
**Pull request by allocom**
Dual mono & Dual Stereo feature added  on Allo Piano Plus

### CODE REVIEW

File: sound/soc/bcm/allo-digione.c [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+		S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP);
+MODULE_PARM_DESC(auto_shutdown_output, "Shutdown SP/DIF output if playback is stopped");
+
+#define CLK_44EN_RATE 22579200UL
+#define CLK_48EN_RATE 24576000UL
+
+static struct gpio_desc *snd_allo_clk44gpio;
+static struct gpio_desc *snd_allo_clk48gpio;
+
+static int samplerate = 44100;
```
Comment (6by9) [inline sound/soc/bcm/allo-digione.c]: Why is this a global when it appears to only be used within snd_allo_digione_hw_params?

File: sound/soc/bcm/allo-piano-dac-plus.c [hunk 2]
```diff
@@ -52,6 +52,8 @@ static const char * const allo_piano_mode_texts[] = {
 	"2.0",
 	"2.1",
 	"2.2",
+	"Dual Stero",
```
Comment (pelwell) [inline sound/soc/bcm/allo-piano-dac-plus.c]: Typo - "Stero".

File: sound/soc/bcm/allo-piano-dac-plus.c [hunk 3]
```diff
@@ -111,13 +113,35 @@ static int __snd_allo_piano_dsp_program(struct snd_soc_pcm_runtime *rtd,
...
 	if (mode == 0) { /* 2.0 */
+		snd_soc_write(rtd->codec_dais[0]->codec,
+				PCM512x_MUTE, 0x00);
 		snd_soc_write(rtd->codec_dais[1]->codec,
 				PCM512x_MUTE, 0x11);
 		glb_ptr->set_rate = rate;
 		glb_ptr->set_mode = mode;
 		glb_ptr->set_lowpass = lowpass;
 		return 1;
+	} else if (mode == 3) { /* dual Stero */
```
Comment (pelwell) [inline sound/soc/bcm/allo-piano-dac-plus.c]: The same typo.

**Review by pelwell — COMMENTED** [review]
Please rebase, address the review comments then resubmit.

File: sound/soc/bcm/allo-digione.c (hunk 1 above)
Comment (allocom) [reply sound/soc/bcm/allo-digione.c]: Thanks for the review, will update.

**Comment by allocom** [comment]
Closing for rebasing & resubmitting.

**Pull request closed without merging**